### PR TITLE
Update notebook 01_pytorch_workflow

### PR DESCRIPTION
**Pull request by beneyal**
- Indent loss calculation for test data in the 1st training loop so it is under inference mode.
- Add general notes in "notes" directory.

### CODE REVIEW

**Comment by mrdbourke** [comment]
Thank you for these, going through now.

Notes wise, do you mind putting these in the pull request for future?

That'll keep everything in place too.

# Notes

> 1. In `LinearRegressionModel`, the `requires_grad` and `dtype` parameters to `torch.randn` are redundant, as `nn.Parameter` takes care of setting them.

Thank you for this, just setting them manually to showcase the use of each argument + defaults, will likely remove for future.

> 2. Again in `LinearRegressionModel`, the `forward` function calculates $w \cdot x + b$, but in the previous notebook, you introduced the linear function in its general form, $x \cdot A^{T} + b$. Since this example is using vectors, it doesn't matter, but it might be worth to note this difference.

Good point, I'm on 01 today.
